### PR TITLE
Fix kw issues

### PR DESCRIPTION
**Pull request by dseah-intel**
- Removed self-assigning code
- Replaced fopen to open

### CODE REVIEW

File: Source/Lib/Codec/EbEncHandle.c [hunk 1]
```diff
@@ -625,33 +627,42 @@ EB_ERRORTYPE EbSetThreadManagementParameters(
...
-                while(*p < '0' || *p > '9') p++;
-                socket_id = strtol(p, NULL, 0);
-                if (socket_id < 0 || socket_id > 15) {
-                    fclose(fin);
-                    return EB_ErrorInsufficientResources;
+    int fd = open("/proc/cpuinfo", O_RDONLY | O_NOFOLLOW, "rt");
+    if (fd > 0) {
+        struct stat file_stat;
+        if (fstat(fd, &file_stat) != -1) {
+            if (S_ISREG(file_stat.st_mode) != 0) {
```
Comment (tianjunwork) [inline Source/Lib/Codec/EbEncHandle.c]: these three if() can be combined to reduce nest if()

Comment (dseah-intel) [reply Source/Lib/Codec/EbEncHandle.c]: the ifs depend on the value in the functions of the previous ifs

Comment (tianjunwork) [reply Source/Lib/Codec/EbEncHandle.c]: struct stat file_stat; 
if(fd >=0 && fstat(fd, &file_stat) ==0 && S_ISREG(file_stat.st_mode) != 0)
with && all the functions are executed.

Comment (dseah-intel) [reply Source/Lib/Codec/EbEncHandle.c]: I did not know that. Will promptly be fixed :)

Comment (dseah-intel) [reply Source/Lib/Codec/EbEncHandle.c]: Done

Comment (dseah-intel) [reply Source/Lib/Codec/EbEncHandle.c]: I realized that if(fd >= 0) needs to be separate or else it would not hit close(fd) when fstat(fd, &file_stat) !=0

**Pull request merged**